woocommerce/woocommerce
# [Store Alerts] Make alert action keys unique

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Follow up to https://github.com/woocommerce/woocommerce/pull/49174
Fixes https://github.com/woocommerce/woocommerce/issues/50437



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

You'll need to make sure you have many notes undeleted and unactioned:

1. Trigger multiple notes. First see what's already in your DB.
```sql
SELECT note_id,is_read,is_deleted,status,type from wp_wc_admin_notes WHERE type = 'update';
```
2. Make more than 10 "unactioned".
```sql
UPDATE wp_wc_admin_notes SET status = 'unactioned' WHERE type = 'update';
```
3. You may need to mark them as not deleted.
```sql 
UPDATE wp_wc_admin_notes SET is_deleted = 0 WHERE type = 'update';
```
4. Scroll through notes on the Homescreen. Make sure the action buttons don't repeat themselves and no `key` warnings appear in the console.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Update Store Alert actions to have unique keys.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
